tests/test_pythia8_fill.py

```diff
@@ -0,0 +1,98 @@
+import chromo
+from chromo.common import EventData
+from .util import run_in_separate_process
+import pytest
+import sys
+import numpy as np
+
+if sys.platform == "win32":
+    pytest.skip("Pythia8 does not run on windows", allow_module_level=True)
+
+
+def init_events(nevents):
+    evt_kin = chromo.kinematics.FixedTarget(1e7, "p", "p")
+    evt_gen = chromo.models.Sibyll23d(evt_kin)
+
+    events = []
+    for event in evt_gen(nevents):
+        event = event.final_state()
+        events.append(event.copy())
+
+    return events
+
+
+def get_pythia_event(pythia, evt_kin):
+    pevent = pythia.event
+    res = EventData(
+        ("pythia", "pythia"),
+        evt_kin,
+        0,
+        0.0,
+        (0, 0),
+        pevent.pid(),
+        pevent.status(),
+        pythia.charge(),
+        pevent.px(),
+        pevent.py(),
+        pevent.pz(),
+        pevent.en(),
+        pevent.m(),
+        pevent.vx(),
+        pevent.vy(),
+        pevent.vz(),
+        pevent.vt(),
+        np.maximum(pevent.mothers() - 1, -1),
+        np.maximum(pevent.daughters() - 1, -1),
+    )
+    return res.copy()
+
+
+def append_result(pythia, event):
+    pythia.event.reset()
+    for i in range(len(event)):
+        pythia.event.append(
+            event.pid[i],
+            91,
+            0,
+            0,
+            event.px[i],
+            event.py[i],
+            event.pz[i],
+            event.en[i],
+            event.m[i],
+        )
+    return get_pythia_event(pythia, event.kin)
+
+
+def fill_result(pythia, event):
```

Comment:
here same "result" sounds not exactly what it does, but since it is only for tests, it might be ok.